File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/fetch_mcp_objects_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class FetchMcpObjectsDetails(object):
    """
    The required details for testing an mcp tool
    """

    #: A constant which can be used with the type property of a FetchMcpObjectsDetails.
    #: This constant has a value of "TOOL"
    TYPE_TOOL = "TOOL"

    #: A constant which can be used with the type property of a FetchMcpObjectsDetails.
    #: This constant has a value of "PROMPT"
    TYPE_PROMPT = "PROMPT"

    #: A constant which can be used with the type property of a FetchMcpObjectsDetails.
    #: This constant has a value of "RESOURCE"
    TYPE_RESOURCE = "RESOURCE"

    #: A constant which can be used with the sort_order property of a FetchMcpObjectsDetails.
    #: This constant has a value of "ASC"
    SORT_ORDER_ASC = "ASC"

    #: A constant which can be used with the sort_order property of a FetchMcpObjectsDetails.
    #: This constant has a value of "DESC"
    SORT_ORDER_DESC = "DESC"

    #: A constant which can be used with the sort_by property of a FetchMcpObjectsDetails.
    #: This constant has a value of "TIME_CREATED"
    SORT_BY_TIME_CREATED = "TIME_CREATED"

    def __init__(self, **kwargs):
        """
        Initializes a new FetchMcpObjectsDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param agent_flow_id:
            The value to assign to the agent_flow_id property of this FetchMcpObjectsDetails.
        :type agent_flow_id: str

        :param type:
            The value to assign to the type property of this FetchMcpObjectsDetails.
            Allowed values for this property are: "TOOL", "PROMPT", "RESOURCE"
        :type type: str

        :param mcp_tool:
            The value to assign to the mcp_tool property of this FetchMcpObjectsDetails.
        :type mcp_tool: oci.aidataplatform_dp.models.McpTool

        :param sort_order:
            The value to assign to the sort_order property of this FetchMcpObjectsDetails.
            Allowed values for this property are: "ASC", "DESC"
        :type sort_order: str

        :param sort_by:
            The value to assign to the sort_by property of this FetchMcpObjectsDetails.
            Allowed values for this property are: "TIME_CREATED"
        :type sort_by: str

        :param limit:
            The value to assign to the limit property of this FetchMcpObjectsDetails.
        :type limit: int

        :param page:
            The value to assign to the page property of this FetchMcpObjectsDetails.
        :type page: str

        :param param_values:
            The value to assign to the param_values property of this FetchMcpObjectsDetails.
        :type param_values: dict(str, str)

        """
        self.swagger_types = {
            'agent_flow_id': 'str',
            'type': 'str',
            'mcp_tool': 'McpTool',
            'sort_order': 'str',
            'sort_by': 'str',
            'limit': 'int',
            'page': 'str',
            'param_values': 'dict(str, str)'
        }

        self.attribute_map = {
            'agent_flow_id': 'agentFlowId',
            'type': 'type',
            'mcp_tool': 'mcpTool',
            'sort_order': 'sortOrder',
            'sort_by': 'sortBy',
            'limit': 'limit',
            'page': 'page',
            'param_values': 'paramValues'
        }

        self._agent_flow_id = None
        self._type = None
        self._mcp_tool = None
        self._sort_order = None
        self._sort_by = None
        self._limit = None
        self._page = None
        self._param_values = None

    @property
    def agent_flow_id(self):
        """
        Gets the agent_flow_id of this FetchMcpObjectsDetails.
        Agent flow id for which the tool is being tested


        :return: The agent_flow_id of this FetchMcpObjectsDetails.
        :rtype: str
        """
        return self._agent_flow_id

    @agent_flow_id.setter
    def agent_flow_id(self, agent_flow_id):
        """
        Sets the agent_flow_id of this FetchMcpObjectsDetails.
        Agent flow id for which the tool is being tested


        :param agent_flow_id: The agent_flow_id of this FetchMcpObjectsDetails.
        :type: str
        """
        self._agent_flow_id = agent_flow_id

    @property
    def type(self):
        """
        **[Required]** Gets the type of this FetchMcpObjectsDetails.
        Type of object. Tool, prompt or resource hosted on an MCP.

        Allowed values for this property are: "TOOL", "PROMPT", "RESOURCE"


        :return: The type of this FetchMcpObjectsDetails.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this FetchMcpObjectsDetails.
        Type of object. Tool, prompt or resource hosted on an MCP.


        :param type: The type of this FetchMcpObjectsDetails.
        :type: str
        """
        allowed_values = ["TOOL", "PROMPT", "RESOURCE"]
        if not value_allowed_none_or_none_sentinel(type, allowed_values):
            raise ValueError(
                "Invalid value for `type`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._type = type

    @property
    def mcp_tool(self):
        """
        **[Required]** Gets the mcp_tool of this FetchMcpObjectsDetails.

        :return: The mcp_tool of this FetchMcpObjectsDetails.
        :rtype: oci.aidataplatform_dp.models.McpTool
        """
        return self._mcp_tool

    @mcp_tool.setter
    def mcp_tool(self, mcp_tool):
        """
        Sets the mcp_tool of this FetchMcpObjectsDetails.

        :param mcp_tool: The mcp_tool of this FetchMcpObjectsDetails.
        :type: oci.aidataplatform_dp.models.McpTool
        """
        self._mcp_tool = mcp_tool

    @property
    def sort_order(self):
        """
        Gets the sort_order of this FetchMcpObjectsDetails.
        The sort order to use, either ascending (`ASC`) or descending (`DESC`). The `displayName`
        sort order is case sensitive.

        Allowed values for this property are: "ASC", "DESC"


        :return: The sort_order of this FetchMcpObjectsDetails.
        :rtype: str
        """
        return self._sort_order

    @sort_order.setter
    def sort_order(self, sort_order):
        """
        Sets the sort_order of this FetchMcpObjectsDetails.
        The sort order to use, either ascending (`ASC`) or descending (`DESC`). The `displayName`
        sort order is case sensitive.


        :param sort_order: The sort_order of this FetchMcpObjectsDetails.
        :type: str
        """
        allowed_values = ["ASC", "DESC"]
        if not value_allowed_none_or_none_sentinel(sort_order, allowed_values):
            raise ValueError(
                "Invalid value for `sort_order`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._sort_order = sort_order

    @property
    def sort_by(self):
        """
        Gets the sort_by of this FetchMcpObjectsDetails.
        The field to sort by.

        Allowed values for this property are: "TIME_CREATED"


        :return: The sort_by of this FetchMcpObjectsDetails.
        :rtype: str
        """
        return self._sort_by

    @sort_by.setter
    def sort_by(self, sort_by):
        """
        Sets the sort_by of this FetchMcpObjectsDetails.
        The field to sort by.


        :param sort_by: The sort_by of this FetchMcpObjectsDetails.
        :type: str
        """
        allowed_values = ["TIME_CREATED"]
        if not value_allowed_none_or_none_sentinel(sort_by, allowed_values):
            raise ValueError(
                "Invalid value for `sort_by`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._sort_by = sort_by

    @property
    def limit(self):
        """
        Gets the limit of this FetchMcpObjectsDetails.
        For list pagination. The maximum number of results per page, or items to return in a
        paginated \"List\" call. For important details about how pagination works, see
        `List Pagination`__.

        __ https://docs.cloud.oracle.com/iaas/Content/API/Concepts/usingapi.htm#nine


        :return: The limit of this FetchMcpObjectsDetails.
        :rtype: int
        """
        return self._limit

    @limit.setter
    def limit(self, limit):
        """
        Sets the limit of this FetchMcpObjectsDetails.
        For list pagination. The maximum number of results per page, or items to return in a
        paginated \"List\" call. For important details about how pagination works, see
        `List Pagination`__.

        __ https://docs.cloud.oracle.com/iaas/Content/API/Concepts/usingapi.htm#nine


        :param limit: The limit of this FetchMcpObjectsDetails.
        :type: int
        """
        self._limit = limit

    @property
    def page(self):
        """
        Gets the page of this FetchMcpObjectsDetails.
        For list pagination. The value of the opc-next-page response header from the previous
        \"List\" call. For important details about how pagination works, see
        `List Pagination`__.

        __ https://docs.cloud.oracle.com/iaas/Content/API/Concepts/usingapi.htm#nine


        :return: The page of this FetchMcpObjectsDetails.
        :rtype: str
        """
        return self._page

    @page.setter
    def page(self, page):
        """
        Sets the page of this FetchMcpObjectsDetails.
        For list pagination. The value of the opc-next-page response header from the previous
        \"List\" call. For important details about how pagination works, see
        `List Pagination`__.

        __ https://docs.cloud.oracle.com/iaas/Content/API/Concepts/usingapi.htm#nine


        :param page: The page of this FetchMcpObjectsDetails.
        :type: str
        """
        self._page = page

    @property
    def param_values(self):
        """
        Gets the param_values of this FetchMcpObjectsDetails.
        Map of parameter names to their string values.


        :return: The param_values of this FetchMcpObjectsDetails.
        :rtype: dict(str, str)
        """
        return self._param_values

    @param_values.setter
    def param_values(self, param_values):
        """
        Sets the param_values of this FetchMcpObjectsDetails.
        Map of parameter names to their string values.


        :param param_values: The param_values of this FetchMcpObjectsDetails.
        :type: dict(str, str)
        """
        self._param_values = param_values

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/FetchMcpObjectsDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * The required details for testing an mcp tool
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=FetchMcpObjectsDetails.Builder.class)

public final class FetchMcpObjectsDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"agentFlowId", "type", "mcpTool", "sortOrder", "sortBy", "limit", "page", "paramValues"})
    public FetchMcpObjectsDetails(String agentFlowId, McpObjectType type, McpTool mcpTool, SortOrder sortOrder, SortBy sortBy, Integer limit, String page, java.util.Map<String, String> paramValues) {
        super();
        this.agentFlowId = agentFlowId;
        this.type = type;
        this.mcpTool = mcpTool;
        this.sortOrder = sortOrder;
        this.sortBy = sortBy;
        this.limit = limit;
        this.page = page;
        this.paramValues = paramValues;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Agent flow id for which the tool is being tested
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("agentFlowId")
private String agentFlowId;

        /**
         * Agent flow id for which the tool is being tested
         * @param agentFlowId the value to set
         * @return this builder
         **/
        

public Builder agentFlowId(String agentFlowId) {
    this.agentFlowId = agentFlowId;
    return this;
}
            /**
     * Type of object. Tool, prompt or resource hosted on an MCP.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private McpObjectType type;

        /**
         * Type of object. Tool, prompt or resource hosted on an MCP.
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(McpObjectType type) {
    this.type = type;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("mcpTool")
private McpTool mcpTool;



public Builder mcpTool(McpTool mcpTool) {
    this.mcpTool = mcpTool;
    return this;
}
            /**
     * The sort order to use, either ascending ({@code ASC}) or descending ({@code DESC}). The {@code displayName}
* sort order is case sensitive.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sortOrder")
private SortOrder sortOrder;

        /**
         * The sort order to use, either ascending ({@code ASC}) or descending ({@code DESC}). The {@code displayName}
* sort order is case sensitive.
* 
         * @param sortOrder the value to set
         * @return this builder
         **/
        

public Builder sortOrder(SortOrder sortOrder) {
    this.sortOrder = sortOrder;
    return this;
}
            /**
     * The field to sort by.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sortBy")
private SortBy sortBy;

        /**
         * The field to sort by.
* 
         * @param sortBy the value to set
         * @return this builder
         **/
        

public Builder sortBy(SortBy sortBy) {
    this.sortBy = sortBy;
    return this;
}
            /**
     * For list pagination. The maximum number of results per page, or items to return in a
* paginated "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("limit")
private Integer limit;

        /**
         * For list pagination. The maximum number of results per page, or items to return in a
* paginated "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
         * @param limit the value to set
         * @return this builder
         **/
        

public Builder limit(Integer limit) {
    this.limit = limit;
    return this;
}
            /**
     * For list pagination. The value of the opc-next-page response header from the previous
* "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("page")
private String page;

        /**
         * For list pagination. The value of the opc-next-page response header from the previous
* "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
         * @param page the value to set
         * @return this builder
         **/
        

public Builder page(String page) {
    this.page = page;
    return this;
}
            /**
     * Map of parameter names to their string values.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("paramValues")
private java.util.Map<String, String> paramValues;

        /**
         * Map of parameter names to their string values.
         * @param paramValues the value to set
         * @return this builder
         **/
        

public Builder paramValues(java.util.Map<String, String> paramValues) {
    this.paramValues = paramValues;
    return this;
}


        public FetchMcpObjectsDetails build() {
            FetchMcpObjectsDetails model = new FetchMcpObjectsDetails(this.agentFlowId
                , this.type
                , this.mcpTool
                , this.sortOrder
                , this.sortBy
                , this.limit
                , this.page
                , this.paramValues);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(FetchMcpObjectsDetails model) {
                this.agentFlowId(model.getAgentFlowId());
    this.type(model.getType());
    this.mcpTool(model.getMcpTool());
    this.sortOrder(model.getSortOrder());
    this.sortBy(model.getSortBy());
    this.limit(model.getLimit());
    this.page(model.getPage());
    this.paramValues(model.getParamValues());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Agent flow id for which the tool is being tested
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowId")
    private final String agentFlowId;

        /**
     * Agent flow id for which the tool is being tested
     * @return the value
     **/
    
    public String getAgentFlowId() {
        return agentFlowId;
    }

    
        /**
     * Type of object. Tool, prompt or resource hosted on an MCP.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final McpObjectType type;

        /**
     * Type of object. Tool, prompt or resource hosted on an MCP.
     * @return the value
     **/
    
    public McpObjectType getType() {
        return type;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("mcpTool")
    private final McpTool mcpTool;

    
    public McpTool getMcpTool() {
        return mcpTool;
    }

    /**
     * The sort order to use, either ascending ({@code ASC}) or descending ({@code DESC}). The {@code displayName}
* sort order is case sensitive.
* 
     **/
    public enum SortOrder implements com.oracle.bmc.http.internal.BmcEnum {
        Asc("ASC"),
        Desc("DESC"),
        ;

        

        private final String value;
        private static java.util.Map<String, SortOrder> map;

        static {
            map = new java.util.HashMap<>();
            for (SortOrder v : SortOrder.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        SortOrder(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static SortOrder create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid SortOrder: " + key);
        }
    };
        /**
     * The sort order to use, either ascending ({@code ASC}) or descending ({@code DESC}). The {@code displayName}
* sort order is case sensitive.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sortOrder")
    private final SortOrder sortOrder;

        /**
     * The sort order to use, either ascending ({@code ASC}) or descending ({@code DESC}). The {@code displayName}
* sort order is case sensitive.
* 
     * @return the value
     **/
    
    public SortOrder getSortOrder() {
        return sortOrder;
    }

    /**
     * The field to sort by.
* 
     **/
    public enum SortBy implements com.oracle.bmc.http.internal.BmcEnum {
        TimeCreated("TIME_CREATED"),
        ;

        

        private final String value;
        private static java.util.Map<String, SortBy> map;

        static {
            map = new java.util.HashMap<>();
            for (SortBy v : SortBy.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        SortBy(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static SortBy create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid SortBy: " + key);
        }
    };
        /**
     * The field to sort by.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sortBy")
    private final SortBy sortBy;

        /**
     * The field to sort by.
* 
     * @return the value
     **/
    
    public SortBy getSortBy() {
        return sortBy;
    }


        /**
     * For list pagination. The maximum number of results per page, or items to return in a
* paginated "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("limit")
    private final Integer limit;

        /**
     * For list pagination. The maximum number of results per page, or items to return in a
* paginated "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
     * @return the value
     **/
    
    public Integer getLimit() {
        return limit;
    }


        /**
     * For list pagination. The value of the opc-next-page response header from the previous
* "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("page")
    private final String page;

        /**
     * For list pagination. The value of the opc-next-page response header from the previous
* "List" call. For important details about how pagination works, see
* [List Pagination]({{DOC_SERVER_URL}}/iaas/Content/API/Concepts/usingapi.htm#nine).
* 
     * @return the value
     **/
    
    public String getPage() {
        return page;
    }


        /**
     * Map of parameter names to their string values.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("paramValues")
    private final java.util.Map<String, String> paramValues;

        /**
     * Map of parameter names to their string values.
     * @return the value
     **/
    
    public java.util.Map<String, String> getParamValues() {
        return paramValues;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("FetchMcpObjectsDetails(");
        sb.append("agentFlowId=").append(String.valueOf(this.agentFlowId));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", mcpTool=").append(String.valueOf(this.mcpTool));
        sb.append(", sortOrder=").append(String.valueOf(this.sortOrder));
        sb.append(", sortBy=").append(String.valueOf(this.sortBy));
        sb.append(", limit=").append(String.valueOf(this.limit));
        sb.append(", page=").append(String.valueOf(this.page));
        sb.append(", paramValues=").append(String.valueOf(this.paramValues));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof FetchMcpObjectsDetails)) {
            return false;
        }

        FetchMcpObjectsDetails other = (FetchMcpObjectsDetails) o;
        return java.util.Objects.equals(this.agentFlowId, other.agentFlowId) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.mcpTool, other.mcpTool) &&
            java.util.Objects.equals(this.sortOrder, other.sortOrder) &&
            java.util.Objects.equals(this.sortBy, other.sortBy) &&
            java.util.Objects.equals(this.limit, other.limit) &&
            java.util.Objects.equals(this.page, other.page) &&
            java.util.Objects.equals(this.paramValues, other.paramValues);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.agentFlowId == null ? 43 : this.agentFlowId.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.mcpTool == null ? 43 : this.mcpTool.hashCode());
        result = (result * PRIME) + (this.sortOrder == null ? 43 : this.sortOrder.hashCode());
        result = (result * PRIME) + (this.sortBy == null ? 43 : this.sortBy.hashCode());
        result = (result * PRIME) + (this.limit == null ? 43 : this.limit.hashCode());
        result = (result * PRIME) + (this.page == null ? 43 : this.page.hashCode());
        result = (result * PRIME) + (this.paramValues == null ? 43 : this.paramValues.hashCode());
        return result;
    }


}
